### PR TITLE
[beta] backports

### PR DESCRIPTION
**Pull request by cuviper**
- Use completion item indices instead of property matching #132987, rust-lang/rust-analyzer#18503
- Reject raw lifetime followed by `'`, like regular lifetimes do #132341
- Only disable cache if predicate has opaques within it #132625
- rustdoc-search: case-sensitive only when capitals are used #133043
- (ci) Update macOS Xcode to 15 #131570

r? cuviper

### CODE REVIEW

File: src/librustdoc/html/static/js/search.js [hunk 1]
```diff
@@ -2097,7 +2097,8 @@ class DocSearch {
          */
         const sortResults = async(results, isType, preferredCrate) => {
             const userQuery = parsedQuery.userQuery;
-            const casedUserQuery = parsedQuery.original;
+            const normalizedUserQuery = parsedQuery.userQuery.toLowerCase();
+            const isMixedCase = normalizedUserQuery !== userQuery;
```
Comment (notriddle) [inline src/librustdoc/html/static/js/search.js:2101]:
```suggestion
            const casedUserQuery = parsedQuery.original;
            const isMixedCase = casedUserQuery !== userQuery;
```

File: src/librustdoc/html/static/js/search.js [hunk 2]
```diff
@@ -2109,15 +2110,17 @@ class DocSearch {
                 let a, b;
 
                 // sort by exact case-sensitive match
-                a = (aaa.item.name !== casedUserQuery);
-                b = (bbb.item.name !== casedUserQuery);
-                if (a !== b) {
-                    return a - b;
+                if (isMixedCase) {
+                    a = (aaa.item.name !== userQuery);
+                    b = (bbb.item.name !== userQuery);
```
Comment (notriddle) [inline src/librustdoc/html/static/js/search.js:2115]:
```suggestion
                    a = (aaa.item.name !== casedUserQuery);
                    b = (bbb.item.name !== casedUserQuery);
```

**Review by notriddle — CHANGES_REQUESTED** [review]

File: src/librustdoc/html/static/js/search.js [hunk 3]
```diff
@@ -2109,15 +2110,17 @@ class DocSearch {
...
+                    if (a !== b) {
+                        return a - b;
+                    }
                 }
 
                 // sort by exact match with regard to the last word (mismatch goes later)
-                a = (aaa.word !== userQuery);
-                b = (bbb.word !== userQuery);
+                a = (aaa.word !== normalizedUserQuery);
+                b = (bbb.word !== normalizedUserQuery);
```
[the remark this answers is not in the document]
Comment (cuviper) [reply src/librustdoc/html/static/js/search.js]: After `./x clean`, it started passing, so I'm not sure what that's about. Let's see if PR CI likes it now...

**Pull request merged**